app/src/main/java/ar/scacchipa/twittercloneapp/presentation/splash/FragmentSplash.kt

```diff
@@ -25,20 +25,27 @@ class FragmentSplash: Fragment() {
         binding = FragmentSplashLayoutBinding.inflate(inflater)
 
         splashViewModel.mustLogin.observe(viewLifecycleOwner) { mustLogin ->
-            if (mustLogin == true) {
-                findNavController().navigate(R.id.action_fragmentSplash_to_fragmentLogin)
-            } else {
-                val intent = Intent(
-                    activity,
-                    MainActivity::class.java
-                )
-                intent.flags =  Intent.FLAG_ACTIVITY_NEW_TASK or
-                        Intent.FLAG_ACTIVITY_CLEAR_TASK
-
-                startActivity( intent )
+            when (mustLogin) {
```

Comment:
add this logic here starts to be messy, views should be as dumm as posible, maybe it is time to start to think in terms of state, and this logic could be moved to viewmodel

Reply:
The StateFlow was implemented in the SpashViewModel.

Reply:
Maybe, it's better to leave these changes for next spring.